File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s580342216_commented.py
# Python translation of the provided Java code

import sys

class Solution:
    def __init__(self):
        self.scanner = sys.stdin

        for line in self.scanner:
            self.run_case(line.strip())

    def run_case(self, line):
        parts = line.split()
        N = int(parts[0])
        K = int(parts[1])

        length = 0
        while N > 0:
            N //= K
            length += 1

        print(length)

    def str_to_int_array(self, str):
        vals = str.split()
        res = [int(val) for val in vals]
        return res


class LCS:
    def __init__(self):
        self.dp = []

    def lcs(self, A, B):
        sz_a = len(A)
        sz_b = len(B)
        self.dp = [[0] * (sz_b + 1) for _ in range(sz_a + 1)]

        for i in range(1, sz_a + 1):
            for j in range(1, sz_b + 1):
                if A[i - 1] == B[j - 1]:
                    self.dp[i][j] = self.dp[i - 1][j - 1] + 1
                else:
                    self.dp[i][j] = max(self.dp[i - 1][j], self.dp[i][j - 1])

        return self.dp[sz_a][sz_b]

    def get_lcs(self, X, Y):
        m = len(X)
        n = len(Y)
        index = self.dp[m][n]
        temp = index

        lcs = [''] * (index + 1)
        lcs[index] = '\0'

        i = m
        j = n
        while i > 0 and j > 0:
            if X[i - 1] == Y[j - 1]:
                lcs[index - 1] = X[i - 1]
                i -= 1
                j -= 1
                index -= 1
            elif self.dp[i - 1][j] > self.dp[i][j - 1]:
                i -= 1
            else:
                j -= 1

        return ''.join(lcs[:temp + 1])

if __name__ == "__main__":
    solution = Solution()

# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/C++/s580342216_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

class Solution {
public:
    Solution() {
        string line;
        while (getline(cin, line)) {
            run_case(line);
        }
    }

private:
    void run_case(const string& line) {
        stringstream ss(line);
        int N, K;
        ss >> N >> K;

        int len = 0;
        while (N > 0) {
            N /= K;
            len++;
        }

        cout << len << endl;
    }

    vector<int> strToIntArray(const string& str) {
        stringstream ss(str);
        vector<int> res;
        int val;
        while (ss >> val) {
            res.push_back(val);
        }
        return res;
    }
};

class LCS {
public:
    int lcs(const string& A, const string& B) {
        int sz_a = A.size(), sz_b = B.size();
        vector<vector<int>> dp(sz_a + 1, vector<int>(sz_b + 1));

        for (int i = 0; i <= sz_a; i++) dp[i][0] = 0;
        for (int j = 0; j <= sz_b; j++) dp[0][j] = 0;

        for (int i = 1; i <= sz_a; i++) {
            for (int j = 1; j <= sz_b; j++) {
                if (A[i - 1] == B[j - 1]) {
                    dp[i][j] = dp[i - 1][j - 1] + 1;
                } else {
                    dp[i][j] = max(dp[i - 1][j], dp[i][j - 1]);
                }
            }
        }

        return dp[sz_a][sz_b];
    }

    string getLCS(const string& X, const string& Y) {
        int m = X.size();
        int n = Y.size();
        int index = dp[m][n];
        int temp = index;

        char lcs[index + 1];
        lcs[index] = '\0';

        int i = m;
        int j = n;
        while (i > 0 && j > 0) {
            if (X[i - 1] == Y[j - 1]) {
                lcs[index - 1] = X[i - 1];
                i--;
                j--;
                index--;
            } else if (dp[i - 1][j] > dp[i][j - 1]) {
                i--;
            } else {
                j--;
            }
        }

        return string(lcs, temp + 1);
    }

private:
    vector<vector<int>> dp;
};

int main() {
    Solution solution;
    // End of Code
